internal/provider/resource_application.go

```diff
@@ -554,6 +587,19 @@ func (r *applicationResource) Create(ctx context.Context, req resource.CreateReq
 		}
 	}
 
+	machines := []string{}
+	if !plan.Machines.IsUnknown() {
+		resp.Diagnostics.Append(plan.Machines.ElementsAs(ctx, &machines, false)...)
+		if resp.Diagnostics.HasError() {
+			return
+		}
+	}
+
+	unitCount := int(plan.UnitCount.ValueInt64())
+	if len(machines) > 0 {
+		unitCount = len(machines)
+	}
```

Comment:
question: isn't this a duplication of what `unitsBasedOnMachineCountModifier` does?

Reply:
removed